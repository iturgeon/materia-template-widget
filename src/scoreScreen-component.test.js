import React from 'react';
import ScoreScreen from './scoreScreen-component';
import renderer from 'react-test-renderer';

describe('ScoreScreen component', function() {

	beforeEach(() => {
		jest.resetModules()
	})

	test('ScoreScreen renders empty table', () => {
		const props = {
			title: 'mock-title',
			scoreTable: []
		}

		const component = renderer.create(<ScoreScreen {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

	});

	test('ScoreScreen renders title and table', () => {
		const props = {
			title: 'mock-title',
			scoreTable: [
				{
					data: [
						"Mock Question 1",
						"Mock Correct Answer 1",
						"Mock Given Answer 1"
					],
					score: 100,
					symbol: "%"
				},
				{
					data: [
						"Mock Question 2",
						"Mock Correct Answer 2",
						"Mock Given Answer 2"
					],
					score: 0,
					symbol: " points"
				}
			]
		}

		const component = renderer.create(<ScoreScreen {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

	});

})
