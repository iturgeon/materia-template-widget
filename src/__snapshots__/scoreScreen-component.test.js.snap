// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreScreen component ScoreScreen renders empty table 1`] = `
<div>
  <div
    id="title"
  >
    mock-title
  </div>
  <div>
    Score Table
  </div>
</div>
`;

exports[`ScoreScreen component ScoreScreen renders title and table 1`] = `
<div>
  <div
    id="title"
  >
    mock-title
  </div>
  <div>
    Score Table
  </div>
  <div>
    <div>
      Question
    </div>
    <div>
      Mock Question 1
    </div>
    <div>
      Score
    </div>
    <div>
      100
      %
    </div>
    <div>
      Correct Answer
    </div>
    <div>
      Mock Correct Answer 1
    </div>
    <div>
      Given Answer
    </div>
    <div>
      Mock Given Answer 1
    </div>
  </div>
  <div>
    <div>
      Question
    </div>
    <div>
      Mock Question 2
    </div>
    <div>
      Score
    </div>
    <div>
      0
       points
    </div>
    <div>
      Correct Answer
    </div>
    <div>
      Mock Correct Answer 2
    </div>
    <div>
      Given Answer
    </div>
    <div>
      Mock Given Answer 2
    </div>
  </div>
</div>
`;
